packages/story-editor/src/components/mediaRecording/footer.js

```diff
@@ -0,0 +1,271 @@
+    countdown: state.countdown,
+    duration: state.duration,
+    isCountingDown: state.isCountingDown,
+    hasMediaToInsert: Boolean(state.mediaBlobUrl),
+    startRecording: actions.startRecording,
+    stopRecording: actions.stopRecording,
+    setFile: actions.setFile,
+    setMediaBlobUrl: actions.setMediaBlobUrl,
+    setDuration: actions.setDuration,
+    setCountdown: actions.setCountdown,
+    toggleRecordingMode: actions.toggleRecordingMode,
+    getMediaStream: actions.getMediaStream,
+    resetState: actions.resetState,
+  }));
+
+  const uploadWithPreview = useUploadWithPreview();
+
+  const onRetry = useCallback(async () => {
+    resetState();
+    await getMediaStream();
+
+    trackEvent('media_recording_retry');
+  }, [getMediaStream, resetState]);
+
+  const onCapture = useCallback(() => {
+    setCountdown(COUNTDOWN_TIME_IN_SECONDS);
+    captureImage();
+
+    trackEvent('media_recording_capture', { type: 'image' });
+  }, [captureImage, setCountdown]);
+
+  const debouncedStartRecording = useDebouncedCallback(startRecording, 3000);
+
+  const onStart = useCallback(() => {
+    setDuration(0);
+    setCountdown(COUNTDOWN_TIME_IN_SECONDS);
+    debouncedStartRecording();
+  }, [debouncedStartRecording, setCountdown, setDuration]);
+
+  const isRecording = 'recording' === status;
+
+  const onInsert = useCallback(async () => {
+    const args = {
+      additionalData: {},
+    };
+    const { resource, posterFile } = await getResourceFromLocalFile(file);
+
+    // Provide video duration
+    args.resource = {
+      ...resource,
+      alt: __('Camera Capture', 'web-stories'),
+    };
+
+    if (file.type.startsWith('video')) {
+      args.additionalData.isMuted = isMuted;
+
+      if (isGif) {
+        args.additionalData.mediaSource = 'gif-conversion';
+      }
+
+      args.resource.length = duration;
+      args.resource.lengthFormatted = getVideoLengthDisplay(duration);
+      args.resource.isMuted = isMuted;
+      args.posterFile = posterFile;
+    }
+
+    uploadWithPreview([file], true, args);
+    toggleRecordingMode();
+
+    // handling cleanup for Image capture
+    // in this case we don't want onStop to override the file
+    if (isRecording) {
+      stopRecording();
+      setMediaBlobUrl(null);
+      setFile(null);
+    }
+
+    trackEvent('media_recording_capture', {
+      type: isGif ? 'gif' : 'video',
+      duration,
+    });
+  }, [
+    file,
+    uploadWithPreview,
+    toggleRecordingMode,
+    isRecording,
+    isGif,
+    duration,
+    isMuted,
+    stopRecording,
+    setMediaBlobUrl,
+    setFile,
+  ]);
+
+  if ('acquiring_media' === status) {
+    return null;
+  }
+
+  return (
+    <StyledFooter showOverflow>
+      {hasMediaToInsert && (
+        <>
+          <RetryButton onClick={onRetry}>
+            {__('Retry', 'web-stories')}
+          </RetryButton>
+          <InsertButton onClick={onInsert}>
+            {__('Insert', 'web-stories')}
```

Comment:
In my testing, the insert can take 1 second to kick in. I think we should disable this button after this is clicked, to ensure that users do not click it more than once.

Reply:
Recording mode is closed within the `onInsert()` callback. The only thing in there that could slow things down is the async `getResourceFromLocalFile()` call.

In my local testing so far that's always been super quick though, even for long videos.

Could you perhaps do some simple time tracking to see how fast/slow it is on your device?

e.g.

```js
// in onInsert
const before = window.performance.now();

// ...
// await getResourceFromLocalFile(file)
// ...

const after = window.performance.now();
console.log(after - before);
```

Reply:
I did some profiling. 

21 second video - 489.49000000208616
2 minute video - 639.6449999958277
2 minute video - 2004.3800000026822 - 4x cpu slowdown. 
1 minute video - 2695.420000001788 - 6x cpu slowdown. 

To me it is noticeable on a 2.3 GHz 8-Core Intel Core i9.

Reply:
https://user-images.githubusercontent.com/237508/174269014-eda3e2b2-ebd3-44da-ad8a-8e66ab2f8b4d.mov

This takes around 3 second for the insert button to disappear. My worry here is that a user will click it twice.

Reply:
Thanks for sharing these numbers and the video 👍 Very helpful